hydrolib/tools/ext_old_to_new/converters.py

```diff
@@ -514,22 +626,42 @@ def convert(
         forcing_model_list = self.convert_tim_to_bc(
             time_model, start_time, units=units, user_defined_names=user_defined_names
         )
+        # set the bc file names to the same names as the tim files.
+        for i, file in enumerate(forcing_model_list):
+            file.filepath = (
+                tim_file.with_stem(f"{tim_file.stem}-{time_model.quantities_names[i]}")
+                .with_suffix(".bc")
+                .name
+            )
+
         data = {
             "id": "L1",
             "name": forcing.quantity,
             "locationfile": location_file,
             "numcoordinates": len(polyline.x),
             "xcoordinates": polyline.x,
             "ycoordinates": polyline.y,
-            "zsource": z_source,
-            "zsink": z_sink,
         }
         forcings = {
             key: value
             for key, value in zip(time_model.quantities_names, forcing_model_list)
         }
         data = data | forcings
-        new_block = SourceSink(**data)
+
+        if None not in z_source:
+            # if the z_source and z_sink are not None, then add them to the data
+            z_source_sink_data = {
+                "zsource": z_source,
+                "zsink": z_sink,
+            }
+            data = data | z_source_sink_data
+
+        try:
+            new_block = SourceSink(**data)
+        except Exception as e:  # pragma: no cover
+            raise SourceSinkError(
+                f"Failed to create the SourceSink object. for the following Errors: {e}"
```

Comment:
```suggestion
                f"Failed to create the SourceSink object, due to the following Errors: {e}"
```
and similar for Meteo etc. (now there was a . in middle of sentence)